src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/TrackedEntitiesBuilder.cs

```diff
@@ -10,16 +12,29 @@ namespace Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.PointsToAnalysis
     /// </summary>
     internal sealed class TrackedEntitiesBuilder : IDisposable
     {
+        /// <summary>
+        /// Stores all the tracked entities.
+        /// NOTE: Entities added to this set should not be removed.
+        /// </summary>
+        private PooledHashSet<AnalysisEntity> _allEntities { get; }
+
         public TrackedEntitiesBuilder()
         {
-            AllEntities = PooledHashSet<AnalysisEntity>.GetInstance();
+            _allEntities = PooledHashSet<AnalysisEntity>.GetInstance();
         }
 
-        public PooledHashSet<AnalysisEntity> AllEntities { get; }
-
         public void Dispose()
         {
-            AllEntities.Free();
+            _allEntities.Free();
         }
+
+        public void AddEntity(AnalysisEntity analysisEntity)
+            => _allEntities.Add(analysisEntity);
+
+        public IEnumerable<AnalysisEntity> EnumerateEntities()
```

Comment:
>EnumerateEntities [](start = 43, length = 17)

Nit: Could make TrackedEntitiesBuilder implement IEnumerable<AnalysisEntity>.

Reply:
This is no longer applicable as now this type tracks both entities and points to values.